From 896ea07f916fc39d6c1a703577f633be2d001d3f Mon Sep 17 00:00:00 2001
From: Mykyta Komarn <nkomarn@hotmail.com>
Date: Thu, 13 Aug 2020 20:45:43 -0700
Subject: [PATCH] Use a shared random for entities


diff --git a/src/main/java/net/minecraft/server/Entity.java b/src/main/java/net/minecraft/server/Entity.java
index 2da303fe..0e70db15 100644
--- a/src/main/java/net/minecraft/server/Entity.java
+++ b/src/main/java/net/minecraft/server/Entity.java
@@ -41,6 +41,21 @@ public abstract class Entity implements ICommandListener {
 
     // CraftBukkit start
     private static final int CURRENT_LEVEL = 2;
+    // Paper start - use a shared random instance
+    public static Random SHARED_RANDOM = new Random() {
+        private boolean locked = false;
+
+        @Override
+        public synchronized void setSeed(long seed) {
+            if (locked) {
+                org.apache.logging.log4j.LogManager.getLogger().error("Ignoring setSeed() on Entity.SHARED_RANDOM", new Throwable());
+            } else {
+                super.setSeed(seed);
+                locked = true;
+            }
+        }
+    };
+    // Paper end
     static boolean isLevelAtLeast(NBTTagCompound tag, int level) {
         return tag.hasKey("Bukkit.updateLevel") && tag.getInt("Bukkit.updateLevel") >= level;
     }
@@ -160,7 +175,7 @@ public abstract class Entity implements ICommandListener {
         this.width = 0.6F;
         this.length = 1.8F;
         this.h = 1;
-        this.random = new Random();
+        this.random = SHARED_RANDOM; // Paper
         this.maxFireTicks = 1;
         this.justCreated = true;
         this.uniqueID = MathHelper.a(this.random);
diff --git a/src/main/java/net/minecraft/server/EntitySquid.java b/src/main/java/net/minecraft/server/EntitySquid.java
index 31996b46..43c94d46 100644
--- a/src/main/java/net/minecraft/server/EntitySquid.java
+++ b/src/main/java/net/minecraft/server/EntitySquid.java
@@ -20,7 +20,7 @@ public class EntitySquid extends EntityWaterAnimal {
     public EntitySquid(World world) {
         super(world);
         this.setSize(0.95F, 0.95F);
-        this.random.setSeed((long) (1 + this.getId()));
+        // this.random.setSeed((long) (1 + this.getId())); // Paper
         this.bq = 1.0F / (this.random.nextFloat() + 1.0F) * 0.2F;
         this.goalSelector.a(0, new EntitySquid.PathfinderGoalSquid(this));
     }
-- 
2.25.1

