From f71833a0bbea3887e0842d332256931c9ef74348 Mon Sep 17 00:00:00 2001
From: Mykyta Komarn <nkomarn@hotmail.com>
Date: Fri, 14 Aug 2020 08:03:43 -0700
Subject: [PATCH] Remove Rabbit carrot goal


diff --git a/src/main/java/net/minecraft/server/EntityRabbit.java b/src/main/java/net/minecraft/server/EntityRabbit.java
index bb47e09a..aa5d1133 100644
--- a/src/main/java/net/minecraft/server/EntityRabbit.java
+++ b/src/main/java/net/minecraft/server/EntityRabbit.java
@@ -34,7 +34,7 @@ public class EntityRabbit extends EntityAnimal {
         this.goalSelector.a(2, new PathfinderGoalTempt(this, 1.0D, Items.GOLDEN_CARROT, false));
         this.goalSelector.a(2, new PathfinderGoalTempt(this, 1.0D, Item.getItemOf(Blocks.YELLOW_FLOWER), false));
         this.goalSelector.a(3, new PathfinderGoalBreed(this, 0.8D));
-        this.goalSelector.a(5, new EntityRabbit.PathfinderGoalEatCarrots(this));
+        // this.goalSelector.a(5, new EntityRabbit.PathfinderGoalEatCarrots(this)); // Paper
         this.goalSelector.a(5, new PathfinderGoalRandomStroll(this, 0.6D));
         this.goalSelector.a(11, new PathfinderGoalLookAtPlayer(this, EntityHuman.class, 10.0F));
         this.bm = new EntityRabbit.PathfinderGoalRabbitAvoidTarget(this, EntityWolf.class, 16.0F, 1.33D, 1.33D);
-- 
2.25.1

